### PR TITLE
CI: Fix meson macos build

### PR DESCRIPTION
**Pull request by andy5995**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,7 +69,7 @@ else
  ifdef CONFIG_OSX
   NATIVE_TARGET = $(ARCH)-osx
   ifneq ($(CC_NAME),tcc)
-    LDFLAGS += -flat_namespace -undefined warning
+    LDFLAGS += -undefined dynamic_lookup
```
Comment (andy5995) [inline Makefile]: This fixed the build using the Makefile. I can't say with confidence it's the best way. I consulted with ChatGPT.

I don't understand yet the problem with the macos meson build. I'll look again later.

Will the Makefiles eventually be removed after #4 is completed?

**Comment by andy5995** [comment]
@KaruroChori based on your reply in https://github.com/KaruroChori/tcc-vs/issues/4#issuecomment-2481683594 do you think I should just remove the macos tests completely? The upstream will be maintaining the macos tests and Makefiles so I wonder if that's just going to be unnecessary work if we do that here.

**Comment by KaruroChori** [comment]
There has been some work which has been done upstream in the last few hours, which replicates (in part) what you have been doing. It looks like the deprecation of images has led many to propagate that change of version :D.

I would be against changing the makefile on `mob` to avoid "collisions", but we can surely have a different branch and prepare it as a patch to send upstream.

As for removing macos tests, I would prefer to have them to be honest, but if they become too much of a problem they can surely go.

**Comment by KaruroChori** [comment]
@andy5995 I just synchronized it with upstream, issues with macos have been mostly fixed by that (except meson on macos 13)

I was thinking, to avoid problems with the `build.yaml` inherited from upstream, we could just move meson built tests in a separate file. This should reduce the frequency of manual resolutions.

**Comment by andy5995** [comment]
> .

Awesome, thank you.

**Comment by KaruroChori** [comment]
According to all the latest running workflows, issues with macos have been fixed. 
It was just a minor patch in a preprocessor directive needed. I also added macos14 to the list, as that one is an aarch64 image.

For reference:  https://repo.or.cz/tinycc.git/commit/ef7e84454ecdc3e892c7f82ad72a2142f9898431

**Comment by andy5995** [comment]
Sweet! Nice work. Thank you for the FYI.

**Pull request closed without merging**